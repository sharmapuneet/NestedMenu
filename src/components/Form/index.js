import React, { useState } from 'react'
import Grid from '@material-ui/core/Grid'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'

const Form = () => {
  const [firstName, setFirstName] = useState('')
  const [lastName, setLastName] = useState('')

  const handleFirstName = event => {
    setFirstName(event.target.value)
  }

  const handleLastName = event => {
    setLastName(event.target.value)
  }

  const handleSubmit = event => {
    event.preventDefault()
    const data = new FormData(document.getElementById('simple-form'))

    data.set('fName', firstName)
    data.set('lName', lastName)

    const settings = {
      method: 'POST',
      body: data,
    }
  
    fetch(`/api/form-submit-url`, settings)

    setFirstName('')
    setLastName('')
    alert(`Form has been successfully submitted with First Name: ${firstName} and Last Name: ${lastName} :)`)
  }

  return (
    <form noValidate autoComplete="off" id="simple-form">
      <Grid container spacing={2}>
        <Grid item xs={12} sm={6}>
          <TextField fullWidth id="fname" label="First Name" value={firstName} onChange={handleFirstName} variant="outlined" />
        </Grid>
        <Grid item xs={12} sm={6}>
          <TextField fullWidth id="lname" label="Last Name" value={lastName} onChange={handleLastName} variant="outlined" />
        </Grid>
      </Grid>
      <Grid container spacing={2}>
        <Grid item xs={12} sm={6}>
          <Button fullWidth variant="contained" color="primary" onClick={handleSubmit}>
            Submit
          </Button>
        </Grid>
        <Grid item xs={12} sm={6}>
          <Button fullWidth variant="contained" color="secondary" href="#/">
            Cancel
          </Button>
        </Grid>
      </Grid>
    </form>
  )
}

export default Form
