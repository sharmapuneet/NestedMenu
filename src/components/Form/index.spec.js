import React from 'react';
import { shallow, render } from 'enzyme';
import Form from './index';

describe('Form', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Form />)
    expect(wrapper).toMatchSnapshot();
  });

  it('should have two textfields', () => {
    const wrapper = render(<Form />)
    expect(wrapper.find('input')).toHaveLength(2);
  });
});