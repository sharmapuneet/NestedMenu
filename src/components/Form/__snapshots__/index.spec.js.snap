// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  autoComplete="off"
  id="simple-form"
  noValidate={true}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        id="fname"
        label="First Name"
        onChange={[Function]}
        value=""
        variant="outlined"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        id="lname"
        label="Last Name"
        onChange={[Function]}
        value=""
        variant="outlined"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        fullWidth={true}
        onClick={[Function]}
        variant="contained"
      >
        Submit
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(Button))
        color="secondary"
        fullWidth={true}
        href="#/"
        variant="contained"
      >
        Cancel
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;
